sandbox/cgroup/cgroup.go

```diff
@@ -259,3 +257,96 @@ func MockVersion(mockVersion int, mockErr error) (restore func()) {
 		probeVersion, probeErr = oldVersion, oldErr
 	}
 }
+
+// securityTagFromCgroupPath returns a security tag from cgroup path.
+func securityTagFromCgroupPath(path string) (securityTag string) {
+	leaf := filepath.Base(filepath.Clean(path))
+	// NOTE: The things we are returning need to minimally look like security
+	// tags. They are further refined and looked at below, in PidsOfSnap.
+	if matched, _ := filepath.Match("snap.*.service", leaf); matched {
+		return strings.TrimSuffix(leaf, ".service")
+	}
+	if matched, _ := filepath.Match("snap.*.scope", leaf); matched {
+		// Neither dot1 nor dot2 can be -1 because the match guarantees that at
+		// least two dots exist.
+		dot1 := strings.IndexRune(leaf, '.')
+		dot2 := strings.IndexRune(leaf[dot1+1:], '.')
+		maybeTag := strings.TrimSuffix(leaf[:dot1]+leaf[dot1+dot2+1:], ".scope")
+		if naming.ValidateSecurityTag(maybeTag) == nil {
+			return maybeTag
+		}
+	}
+	return ""
+}
+
+// PidsOfSnap returns the association of security tags to PIDs.
+//
+// NOTE: This function returns non-empty result only if refresh-app-awareness
+// is enabled.
```

Comment:
Good catch, the comment is out of date with reality here. Since services do not require this to be tracked the comment is inaccurate. I'll correct it.

Reply:
I fixed the comment and added a TODO to consider not checking for the pids at all if the feature is disabled. I think it makes more sense to return consistent results than no results at all. Having said that I would like to change this in the next iteration as it may have additional consequences that I don't want to drag into this (changes to refresh.go)